packages/plugins/scripts-runner/src/index.ts

```diff
@@ -0,0 +1,223 @@
+        this._block = await web3.eth.getBlock(1, true);
+      }
+      return this._block;
+    })();
+  }
+
+  private async execute(target: string, forceTracking = false, callback: Callback<any>) {
+    const targetPath = !path.isAbsolute(target) ? dappPath(target) : target;
+    try {
+      const fstat = await fs.stat(targetPath);
+      if (fstat.isDirectory()) {
+        const dependencies = await this.getScriptDependencies();
+        const results = await this.executeAll({ target: targetPath, dependencies, forceTracking });
+        callback(null, results);
+      } else if (fstat.isFile()) {
+        const dependencies = await this.getScriptDependencies();
+        const result = await this.executeSingle({ target: targetPath, dependencies, forceTracking });
+        callback(null, result);
+      } else {
+        callback(new UnsupportedTargetError(fstat));
+      }
+    } catch (e) {
+      callback(e);
+    }
+  }
+
+  private async executeSingle(options: ExecuteOptions) {
+    const forceTracking = options.forceTracking;
+    const scriptName = path.basename(options.target);
+    const scriptDirectory = path.basename(path.dirname(options.target));
+    const scriptTracked = await this.tracker.isTracked(scriptName);
+
+    if (scriptTracked && (scriptDirectory === this.embark.config.embarkConfig.migrations || forceTracking)) {
+      this.embark.logger.info(`  ✓ ${scriptName} already done`);
+      return;
+    }
+
+    this.embark.logger.info(`  ${scriptName} running....`);
+    const scriptToRun = require(options.target);
+
+    let result;
+
+    try {
+      result = await scriptToRun(options.dependencies);
+    } catch (e) {
+      const error = e instanceof Error ? e : new Error(e);
+      throw new ScriptExecutionError(options.target, error);
+    }
+
+    this.embark.logger.info(`  ✓ finished.`);
+
+    if (!this.trackingEnabled) {
+      return result;
+    }
+
+    try {
+      await this.tracker.track({
+        scriptName,
+        scriptDirectory,
+        forceTracking
+      });
+    } catch (e) {
+      const error = e instanceof Error ? e : new Error(e);
+      throw new ScriptTrackingError(e);
+    }
+    return result;
+  }
+
+  private async executeAll(options: ExecuteOptions) {
+    const target = options.target;
+    const files = await fs.readdir(target);
+
+    const scripts = await Promise.all(
+      files.map(async file => {
+        const targetPath = !path.isAbsolute(target) ? dappPath(target, file) : path.join(target, file);
+        return { target: targetPath, stats: await fs.stat(targetPath) };
+      })
+    );
+
+    return AsyncIterator.from(scripts)
+      .filter(({stats}) => stats.isFile())
+      .map(fstat => ({
+        target: fstat.target,
+        dependencies: options.dependencies,
+        forceTracking: options.forceTracking
+      }))
+      .map(script => this.executeSingle(script))
+      .toArray();
+  }
```

Comment:
@michaelsbradleyjr This is your revised version of `executeAll()` - works like a charm!